src/backend/code.d

```diff
@@ -0,0 +1,40 @@
+/**
+ * Compiler implementation of the
+ * $(LINK2 http://www.dlang.org, D programming language).
+ *
+ * Copyright:   Copyright (C) 1985-1996 by Symantec
+ *              Copyright (c) 2000-2016 by Digital Mars, All Rights Reserved
+ * Authors:     $(LINK2 http://www.digitalmars.com, Walter Bright)
+ * License:     backendlicense.txt
+ * Source:      $(DMDSRC backend/_code.d)
+ */
+
+module ddmd.backend.code;
+
+import ddmd.backend.cdef;
+import ddmd.backend.type;
+
+extern (C++):
+
+/**************************************************/
+
+/* Allocate registers to function parameters
+ */
+
+struct FuncParamRegs
+{
+    //this(tym_t tyf);
+    static FuncParamRegs create(tym_t tyf);
+
+    int alloc(type *t, tym_t ty, ubyte *reg1, ubyte *reg2);
+
+  private:
+    tym_t tyf;                  // type of function
```

Comment:
When doing translations, I try hard to do as few changes as possible. I don't do any fixes, change comments, etc. I just want to get it to work - doing as few changes as possible makes it much, much easier to track down where a translation went wrong.

Fixes, improvements, converting to D idiomatic constructs, etc., all come after it is working.